crates/types/src/traits/auction_results.rs

```diff
@@ -0,0 +1,31 @@
+//! This module defines the interaction layer with the Solver via the [`AuctionResults`] trait,
+//! which handles connecting to, and fetching the allocation results from, the Solver.
+
+use super::node_implementation::NodeType;
+use anyhow::Result;
+use async_trait::async_trait;
+use url::Url;
+
+/// This trait guarantees that a particular type has a url associated with it. This trait
+/// essentially ensures that the results returned by the [`AuctionResults`] trait includes a URL
+/// for the builder that HotShot must request from.
+pub trait HasUrl {
+    /// Returns the builer url associated with the datatype
+    fn url(&self) -> Url;
+}
+
+/// The AuctionResults trait is the sole source of Solver-originated allocations, and returns
+/// the results via its own custom type.
+#[async_trait]
+pub trait AuctionResults<TYPES: NodeType>: Send + Sync {
```

Comment:
I find the name of this trait a little confusing, especially with the associated type `AuctionSolverResult`: it's not totally clear what the difference is. May I suggest calling this something like `AuctionProvider` or `AuctionSolver`, since this trait is really an interface to _get_ the results, note the results themselves? Then we could just call the associated type `AuctionResult`

Reply:
Done.